### PR TITLE
feat: update profile info api

### PR DESCRIPTION
**Pull request by datphamcode295**
#### What's this PR do?

- [x] Update profile info api

#### What are the relevant Git tickets?
https://www.notion.so/API-to-edit-profile-information-b1e05cfc78444a1887a33797926d99f8

#### Screenshots (if appropriate)
image

#### Request
```
curl 'http://localhost:8200/api/v1/profile/info'
-X PUT
-H 'accept: /'
-H 'accept-language: vi,en-US;q=0.9,en;q=0.8'
-H 'content-type: application/json'
-H 'origin: http://localhost:3000/'
-H 'referer: http://localhost:3000/'
-H 'sec-ch-ua: "Chromium";v="106", "Google Chrome";v="106", "Not;A=Brand";v="99"'
-H 'sec-ch-ua-mobile: ?0'
-H 'sec-ch-ua-platform: "macOS"'
-H 'sec-fetch-dest: empty'
-H 'sec-fetch-mode: cors'
-H 'sec-fetch-site: cross-site'
-H 'user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/106.0.0.0 Safari/537.36'
-H 'authorization:Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE2OTkzMjExNDIsImlkIjoiMjY1NTgzMmUtZjAwOS00YjczLWE1MzUtNjRjM2EyMmU1NThmIiwiYXZhdGFyIjoiaHR0cHM6Ly9zMy1hcC1zb3V0aGVhc3QtMS5hbWF6b25hd3MuY29tL2ZvcnRyZXNzLWltYWdlcy81MTUzNTc0Njk1NjYzOTU1OTQ0LnBuZyIsImVtYWlsIjoidGhhbmhAZC5mb3VuZGF0aW9uIiwicGVybWlzc2lvbnMiOlsiZW1wbG95ZWVzLnJlYWQiXSwidXNlcl9pbmZvIjpudWxsfQ.GENGPEucSUrILN6tHDKxLMtj0M0REVMUPC7-XhDMpGM'
--data-raw '{
   "email": "thanh@d.foundation",
   "phone":"0123456788"
}'
--compressed
```

#### Response
```
{
    "data": {
        "id": "2655832e-f009-4b73-a535-64c3a22e558f",
        "createdAt": "2022-11-02T09:52:34.586566Z",
        "updatedAt": "2022-11-16T23:28:30.257166Z",
        "teamEmail": "thanh@d.foundation",
        "phoneNumber": "0123456788",
        "discordID": "",
        "githubID": "",
        "notionID": ""
    }
}
```

### CODE REVIEW

File: pkg/routes/v1.go [hunk 1]
```diff
@@ -15,7 +15,8 @@ func loadV1Routes(r *gin.Engine, h *handler.Handler, repo store.DBRepo, s *store
 	v1.POST("/auth", h.Auth.Auth)
 
 	// user profile
-	v1.GET("/profile", mw.WithAuth, h.Employee.GetProfile)
+	v1.GET("/profile", mw.WithAuth, h.Profile.GetProfile)
+	v1.PUT("/profile/info", mw.WithAuth, h.Profile.UpdateInfo)
```
Comment (nguyennh4522) [inline pkg/routes/v1.go:19]:
```suggestion
	v1.PUT("/profile", mw.WithAuth, h.Profile.UpdateInfo)
```

File: pkg/handler/profile/profile_test.go [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+			ctx.Request.Header.Set("Authorization", "Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE2OTkzMjExNDIsImlkIjoiMjY1NTgzMmUtZjAwOS00YjczLWE1MzUtNjRjM2EyMmU1NThmIiwiYXZhdGFyIjoiaHR0cHM6Ly9zMy1hcC1zb3V0aGVhc3QtMS5hbWF6b25hd3MuY29tL2ZvcnRyZXNzLWltYWdlcy81MTUzNTc0Njk1NjYzOTU1OTQ0LnBuZyIsImVtYWlsIjoidGhhbmhAZC5mb3VuZGF0aW9uIiwicGVybWlzc2lvbnMiOlsiZW1wbG95ZWVzLnJlYWQiXSwidXNlcl9pbmZvIjpudWxsfQ.GENGPEucSUrILN6tHDKxLMtj0M0REVMUPC7-XhDMpGM")
+			metadataHandler := New(storeMock, testRepoMock, serviceMock, loggerMock)
+
+			metadataHandler.UpdateInfo(ctx)
+
+			require.Equal(t, tt.wantCode, w.Code)
+			expRespRaw, err := ioutil.ReadFile(tt.wantResponsePath)
+			require.NoError(t, err)
+
+			if !tt.wantErr {
```
Comment (nguyennh4522) [inline pkg/handler/profile/profile_test.go:117]:
```suggestion
			res, err := utils.RemoveFieldInResponse(w.Body.Bytes(), "updatedAt")
			require.JSONEq(t, string(expRespRaw), string(res), "[Handler.UpdateInfo] response mismatched")
```

Comment (nguyennh4522) [reply pkg/handler/profile/profile_test.go:117]: we needn't check err here

File: pkg/handler/profile/profile.go [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+func (h *handler) UpdateInfo(c *gin.Context) {
+	employeeID, err := utils.GetUserIDFromContext(c)
+
+	if err != nil {
+		c.JSON(http.StatusBadRequest, view.CreateResponse[any](nil, nil, err, nil))
+		return
+	}
+
+	input := UpdateInforInput{}
+	pp.Println(input)
```
Comment (nguyennh4522) [inline pkg/handler/profile/profile.go:98]: remove log if its not necessary

Comment (nguyennh4522) [reply pkg/handler/profile/profile.go:98]: should be log by `h.logger`

File: pkg/routes/v1.go (hunk 1 above)
Comment (datphamcode295) [reply pkg/routes/v1.go:19]: we have another API to update the avatar, so should I remove the "/info" @namnhce

File: pkg/handler/profile/profile_test.go (hunk 2 above)
Comment (datphamcode295) [reply pkg/handler/profile/profile_test.go:117]: updated

File: pkg/handler/profile/profile.go (hunk 3 above)
Comment (datphamcode295) [reply pkg/handler/profile/profile.go:98]: updated

File: pkg/handler/profile/profile.go [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+// @Router /profile/info [put]
+func (h *handler) UpdateInfo(c *gin.Context) {
+	employeeID, err := utils.GetUserIDFromContext(c)
+
+	if err != nil {
+		c.JSON(http.StatusBadRequest, view.CreateResponse[any](nil, nil, err, nil))
+		return
+	}
+
+	input := UpdateInforInput{}
```
Comment (namnhce) [inline pkg/handler/profile/profile.go:96]:
```suggestion
	input := UpdateInfoInput{}
```

File: pkg/handler/profile/request.go [hunk 5]
```diff
@@ -0,0 +1,10 @@
+package profile
+
+// UpdateInforInput input model for update profile
+type UpdateInforInput struct {
+	Email     string `form:"email" json:"email" binding:"required,email"`
+	Phone     string `form:"phone" json:"phone" binding:"required,max=12,min=10"`
+	DiscordID string `form:"discordID" json:"discordID"`
+	GithubID  string `form:"githubID" json:"githubID"`
+	NotionID  string `form:"notionID" json:"notionID"`
```
Comment (namnhce) [inline pkg/handler/profile/request.go:9]:
```suggestion
	TeamEmail     string `form:"email" json:"teamEmail" binding:"required,email"`
	PhoneNumber     string `form:"phone" json:"phoneNumber" binding:"required,max=12,min=10"`
	DiscordID string `form:"discordID" json:"discordID"`
	GithubID  string `form:"githubID" json:"githubID"`
	NotionID  string `form:"notionID" json:"notionID"`
```

File: pkg/routes/v1.go (hunk 1 above)
Comment (namnhce) [reply pkg/routes/v1.go:19]: Yes.

File: pkg/store/employee/request.go [hunk 6]
```diff
@@ -34,3 +34,11 @@ type UpdatePersonalInfoInput struct {
 	Address       string
 	PersonalEmail string
 }
+
+type UpdateProfileInforInput struct {
+	Email     string
+	Phone     string
+	DiscordID string
+	GithubID  string
+	NotionID  string
```
Comment (namnhce) [inline pkg/store/employee/request.go:43]:
```suggestion
	TeamEmail     string
	PhoneNumber     string
	DiscordID string
	GithubID  string
	NotionID  string
```

**Review by namnhce — CHANGES_REQUESTED** [review]

File: pkg/handler/profile/profile.go (hunk 4 above)
Comment (datphamcode295) [reply pkg/handler/profile/profile.go:96]: updated

File: pkg/handler/profile/request.go (hunk 5 above)
Comment (datphamcode295) [reply pkg/handler/profile/request.go:9]: updated

File: pkg/store/employee/request.go (hunk 6 above)
Comment (datphamcode295) [reply pkg/store/employee/request.go:43]: updated

File: pkg/handler/profile/profile.go [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+	"errors"
+	"net/http"
+
+	"github.com/dwarvesf/fortress-api/pkg/logger"
+	"github.com/dwarvesf/fortress-api/pkg/service"
+	"github.com/dwarvesf/fortress-api/pkg/store"
+	"github.com/dwarvesf/fortress-api/pkg/store/employee"
+	"github.com/dwarvesf/fortress-api/pkg/utils"
+	"github.com/dwarvesf/fortress-api/pkg/view"
+	"github.com/gin-gonic/gin"
```
Comment (namnhce) [inline pkg/handler/profile/profile.go:13]: rearrange import

File: pkg/handler/profile/profile.go [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+// @Accept  json
+// @Produce  json
+// @Param Authorization header string true "jwt token"
+// @Param id path string true "Employee ID"
+// @Param Body body UpdateInfoInput true "Body"
+// @Success 200 {object} view.UpdateProfileInfoResponse
+// @Failure 400 {object} view.ErrorResponse
+// @Failure 404 {object} view.ErrorResponse
+// @Failure 500 {object} view.ErrorResponse
+// @Router /profile/info [put]
```
Comment (namnhce) [inline pkg/handler/profile/profile.go:87]: Please change this line

**Review by namnhce — CHANGES_REQUESTED** [review]

File: pkg/handler/profile/profile.go (hunk 7 above)
Comment (datphamcode295) [reply pkg/handler/profile/profile.go:13]: updated

File: pkg/handler/profile/profile.go (hunk 8 above)
Comment (datphamcode295) [reply pkg/handler/profile/profile.go]: updated

**Review by namnhce — APPROVED** [review]

**Pull request merged**